src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -54,17 +54,15 @@ std::vector<TRShape> shape_infer(const Squeeze* op,
                 unique_axes.reset(new std::set<int64_t>(axes->cbegin(), axes->cend()));
             } else if (arg_rank.get_length() > 0 && shape_size(axes_shape.to_shape()) == 1) {
                 // The `axes` input is a single element tensor which is unique by definition, deducing output rank
-                NODE_VALIDATION_CHECK(op,
-                                      std::any_of(arg_shape.cbegin(),
-                                                  arg_shape.cend(),
-                                                  [](const DimType& dim) {
-                                                      return dim.compatible(1);
-                                                  }),
-                                      "Data input shape ",
-                                      arg_shape,
-                                      " doesn't contain squeezable dimension,"
-                                      " but axes input is expected to have one element.");
-                output_shape = PartialShape::dynamic(arg_rank.get_length() - 1);
+                const auto has_squeezable_dim =
+                    std::any_of(arg_shape.cbegin(), arg_shape.cend(), [](const DimType& dim) {
+                        return dim.compatible(1);
+                    });
+                if (has_squeezable_dim) {
+                    output_shape = PartialShape::dynamic(arg_rank.get_length() - 1);
+                } else {
+                    output_shape = arg_shape;
+                }
```

Comment:
in the new specification, this case is incorrect. we can not predict if this dimension will be squeezed or not. it should return dynamic output shape

Reply:
That behavior is preserved for backward compatibility, this is how it worked before, and changing it to dynamic rank can break currently supported models. This PR unblocked the cases with index of definitely non-squeezable (`!compatible(1)`) dims, causing an error in the previous version. I agree that producing dynamic rank would be more suitable for new cases, but to enable the cases where dynamic dimension compatible(1) can be also non-squeezable at the end, we need to find a solution to resolve dynamic rank and keep backward compatibility.